pennylane/ops/qubit/hamiltonian.py

```diff
@@ -86,6 +91,7 @@ class Hamiltonian(Observable):
     A Hamiltonian can be created by simply passing the list of coefficients
     as well as the list of observables:
 
+    >>> qml.operation.disable_new_opmath()
```

Comment:
Instead, could the outputs here be updated to what happens when the new opmath is enabled?

Reply:
I was also thinking about that, but wouldnt that be more confusing than helping? This is the legacy Hamiltonian class, and we would be showing examples for the new LinearCombination class, in the docs of the legacy Hamiltonian class.

Reply:
Perhaps we should long term be thinking of something more informative on top level `qml.Hamiltonian` doc. With this solution here, should it turn out to work, we _could_ potentially generate a dummy `Hamiltonian` class that just points to `Hamiltonian` and `LinearCombination` docs and explains the dispatching. The problem here is that actually `pennylane.ops.Hamiltonian` is not rendered anywhere right now (we only render the `ops.op_math` classes and functions in the docs